service/s3/s3manager/upload.go

```diff
@@ -480,19 +501,33 @@ func readFillBuf(r io.Reader, b []byte) (offset int, err error) {
 // a regular PutObject request. Multipart requests require at least two
 // parts, or at least 5MB of data.
 func (u *uploader) singlePart(buf io.ReadSeeker) (*UploadOutput, error) {
+	var (
+		r       io.ReadSeeker
+		cleanup func()
+	)
+
+	r = buf
+	if u.cfg.UploadBufferStrategy != nil {
+		r, cleanup = u.cfg.UploadBufferStrategy.GetWriteTo(r)
+	} else {
+		cleanup = func() {}
+	}
+
```

Comment:
instead of having `cleanup` called explicitly below, what do you think about having it as a `defer` after the above if/else block where it is set?